client/src/main/java/org/evosuite/testcase/TestFactory.java

```diff
@@ -2281,24 +2313,44 @@ public int insertRandomStatement(TestCase test, int lastPosition) {
 	 *
 	 * @param test
 	 * @param parameterTypes
+	 * @param parameterList
 	 * @param position
 	 * @param recursionDepth
 	 * @return
 	 * @throws ConstructionFailedException
 	 */
-	public List<VariableReference> satisfyParameters(TestCase test,
-	        VariableReference callee, List<Type> parameterTypes, int position,
-	        int recursionDepth, boolean allowNull, boolean excludeCalleeGenerators,
-													 boolean canReuseExistingVariables) throws ConstructionFailedException {
+	public List<VariableReference> satisfyParameters(TestCase test, VariableReference callee, List<Type> parameterTypes,
+			List<Parameter> parameterList, int position, int recursionDepth, boolean allowNull,
+			boolean excludeCalleeGenerators, boolean canReuseExistingVariables) throws ConstructionFailedException {
 
-		if(callee==null && excludeCalleeGenerators){
+		if (callee == null && excludeCalleeGenerators) {
 			throw new IllegalArgumentException("Exclude generators on null callee");
 		}
 
 		List<VariableReference> parameters = new ArrayList<>();
-		logger.debug("Trying to satisfy {} parameters at position {}",parameterTypes.size(),position);
-
-		for (Type parameterType : parameterTypes) {
+		logger.debug("Trying to satisfy {} parameters at position {}", parameterTypes.size(), position);
+
+		// changes start
+		List<?> genericParameterList = new ArrayList<>();
+		Boolean isListOfParameterType = true;
+		if (null != parameterList) {
+			genericParameterList = parameterList;
+			isListOfParameterType = false;
+		} else {
+			genericParameterList = parameterTypes;
+		}
+		for (Object genericParameter : genericParameterList) {
```

Comment:
Wouldn't it be simpler to iterate over the number of parameters (for(int i = 0; i < parameterTypes.size(); i++)) and then retrieve the item at that index? That could avoid some of these additional if/else constructs